### DISCUSSION TITLE
Suggestion: Add mulitcast support for QLog  WSJTX UDP RX port

### DISCUSSION THREAD

**Discussion by rsaue** [foldynl/QLog#171] [Ideas]
Hi Ladislaw, 
The wsjtx integration with QLog is nice, and  QLog has great port forwarding options. 

But as far as I understand QLog does not support multicast. 

Without muliticast only QLog can communicate directly with wsjtx/jtdx. Users using Gridtracker https://www.gridtracker.org can then only monitor UDP packets forwarded from QLog, but cannot communicate directly with wsjtx/jtdx. 

If QLog add support for multicast, this woud be possible. 

(I know that Gridtracker can be configured with port forwarding like QLog, but then Qlog would lose its capability to give instructions directly to wsjtx/jtdx.)

73 de LB2EG Richard

**Comment by foldynl** [comment]
To be honest, I thought that multicast would not be used much - there are certain specifications for multicast that suppress this type of traffic. But you're right. If it is possible to use it in WSTJX, then it is necessary to implement this as well.

**Reply by rsaue** [reply]
Thank you for mulitcast. 

Tested and works in my shack.  

Richard

**Reply by foldynl** [reply]
Many thanks for feedback. I'm glad it works. It wasn't very clear to me how WSJTX works, so I had to look at its source code. Hopefully it will be reliable. It is funny that WSJTX sends messages via multicast but it does not receive commands via Multicast. That was surprising to me. Another thing you need to pay attention to is loopback interface. I don't have multicast enabled in my Linux by default.

**Reply by rsaue** [reply]
Hi, 
using MSHV, multicast is a two way street. I also believe this also is the case with JTDX. 

Richard

**Reply by rsaue** [reply]
In my setup WSJTX / JTDX /MSHV also receive commands via multicast. I am using  multicast adress 239.255.xx.xx and port 2239 in WSJTX, Gridtracker, CQRLOG and QLog, and all programs send and receive commands via Multicast. I can start a QSO in WSJTX by clicking on a call sign in QLog wsjtx window. When testing I was using WSJT-X v 2.6.1. 

So far all good!

73 de LB2EG Richard
